src/main/kotlin/com/hypto/iam/server/Application.kt

```diff
@@ -137,6 +142,29 @@ fun Application.handleRequest() {
     install(HSTS, applicationHstsConfiguration()) // see http://ktor.io/features/hsts.html
     install(Compression, applicationCompressionConfiguration()) // see http://ktor.io/features/compression.html
     install(Authentication) {
+        oauth("oauth") {
+            validate {
+                when (this.request.headers["issuer"]) {
+                    "google" -> {
+                        val httpClient = OkHttpClient()
+                        val requestBuilder = Request.Builder()
+                            .url("https://www.googleapis.com/oauth2/v3/userinfo?access_token=${it.value}")
+                            .method("GET", null)
+                            .addHeader("Content-Type", "application/json")
+                            .addHeader("Connection", "keep-alive")
+                        val request = requestBuilder.build()
+                        val response = httpClient.newCall(request).execute()
+                        if (!response.isSuccessful) {
+                            throw BadRequestException("Invalid token")
```

Comment:
> Should we log / return failure reason?

Yes, I will log

> shouldn't we return 401?

Yes, we should throw `AuthenticationException` rather than `BadRequestException`